source/Octopus.Tentacle/Kubernetes/KubernetesModule.cs

```diff
@@ -15,6 +15,13 @@ protected override void Load(ContainerBuilder builder)
         {
             builder.RegisterType<KubernetesPodService>().As<IKubernetesPodService>().SingleInstance();
             builder.RegisterType<KubernetesClusterService>().As<IKubernetesClusterService>().SingleInstance();
+
+            const string agentToolsImageMetadataProvider = "AgentToolsImageMetadataProvider";
+            builder.RegisterType<KubernetesAgentToolsImageVersionMetadataProvider>().Named<IToolsImageVersionMetadataProvider>(agentToolsImageMetadataProvider);
+            builder.RegisterDecorator<IToolsImageVersionMetadataProvider>(
+                (context, inner) => new CachingKubernetesAgentToolsImageVersionMetadataProvider(
+                    inner, context.Resolve<IMemoryCache>()), fromKey: agentToolsImageMetadataProvider);
```

Comment:
```suggestion
            builder.RegisterDecorator<CachingKubernetesAgentToolsImageVersionMetadataProvider, IToolsImageVersionMetadataProvider>();
```
You _should_ be able to just do this and not need the named service?

Reply:
I tried that at first but was getting a compiler error. It seems that this syntax was introduced in Autofac [version 4.9](https://autofac.readthedocs.io/en/latest/advanced/adapters-decorators.html#simplified-syntax). Tentacle is on version 4.6.2

Reply:
_sigh_ of course 😢